lib/crypto/packaged.py

```diff
@@ -51,6 +52,17 @@ def get_endpoint(server):
     return u'{server}/1.0/sign_addon'.format(server=server)
 
 
+def get_id(addon):
+    """Return the addon GUID if <= 64 chars, or its sha256 hash otherwise.
+
+    We don't want GUIDs longer than 64 chars: bug 1203365.
+    """
+    guid = addon.guid
+    if len(guid) <= 64:
+        return guid
+    return hashlib.sha256(guid).hexdigest()
```

Comment:
Yes, I believe the client will check the ID length from the install.rdf, just as we do, and then decide whether it's the hash of the ID itself.
If we need to add a prefix, then the length will be over 64 chars, and then we have the same issue ;)